test/jdk/java/awt/Focus/8000326/SetFocusTraversalKeysEnabledTest.java

```diff
@@ -0,0 +1,108 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8000326
+ * @key headful
+ * @summary Focus unable to traverse within the menubar
+ * @run main SetFocusTraversalKeysEnabledTest
+ */
+import java.awt.AWTException;
+import java.awt.BorderLayout;
+import java.awt.Component;
+import java.awt.ContainerOrderFocusTraversalPolicy;
+import java.awt.Robot;
+import java.awt.event.KeyEvent;
+import java.lang.reflect.InvocationTargetException;
+
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JMenu;
+import javax.swing.JMenuBar;
+import javax.swing.JMenuItem;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+
+public class SetFocusTraversalKeysEnabledTest {
+
+    private static volatile JFrame jFrame;
+    private static volatile Component currentFocusOwner;
+    private static Robot robot;
+
+    private static void doTest()
+        throws InvocationTargetException, InterruptedException, AWTException {
+        try {
+            SwingUtilities.invokeAndWait(() -> createGUI());
+            Robot robot = new Robot();
+            robot.setAutoDelay(500);
+            Component lastFocusOwner = null;
+            do {
+                robot.waitForIdle();
+                SwingUtilities.invokeAndWait(() -> {
+                    currentFocusOwner = jFrame.getFocusOwner();
+                });
+                System.out.println("Focus owner is : " + currentFocusOwner.getClass().getName());
+                if (currentFocusOwner == lastFocusOwner) {
+                    throw new RuntimeException(
+                        "Problem moving focus from " + currentFocusOwner.getClass().getName());
+                }
+                lastFocusOwner = currentFocusOwner;
+                robot.keyPress(KeyEvent.VK_TAB);
+                robot.keyRelease(KeyEvent.VK_TAB);
+            } while (currentFocusOwner != jFrame);
+        } finally {
+            SwingUtilities.invokeAndWait(() -> {
+                jFrame.dispose();
+            });
+        }
+    }
+
+    private static void createGUI() {
+        jFrame = new JFrame("Focus Traversal Test");
+        jFrame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+
+        JMenuBar jMenuBar = new JMenuBar();
+        jMenuBar.setFocusTraversalKeysEnabled(true);
+        jMenuBar.add(new JMenu("First Menu").add(new JMenuItem("First MenuItem")));
+
+        JButton northButton = new JButton("North Button");
+        JButton southButton = new JButton("South Button");
+
+        JPanel jPanel = new JPanel(new BorderLayout());
+        jPanel.add(northButton);
+        jPanel.add(jMenuBar, BorderLayout.NORTH);
+        jPanel.add(southButton, BorderLayout.SOUTH);
+
+        jFrame.getContentPane().add(jPanel);
+        jFrame.setFocusTraversalPolicy(new ContainerOrderFocusTraversalPolicy());
+        jFrame.pack();
+        northButton.requestFocusInWindow();
+        jFrame.setVisible(true);
```

Comment:
You can add
```java
        jFrame.setLocationRelativeTo(null);
```
before calling `setVisible` to show the frame in the center of the screen, which is the recommended location for tests.